scripts/create-kubestellar-demo-env.sh

```diff
@@ -84,35 +117,62 @@ checking_cluster() {
 
     done
 }
+
+# Function to check if a command exists
+command_exists() {
+    command -v "$1" >/dev/null 2>&1
+}
 ##########################################
 
 echo -e "\nStarting environment clean up..."
 echo -e "Starting cluster clean up..."
 
-cluster_clean_up "kind delete cluster --name kubeflex" &
-cluster_clean_up "kind delete cluster --name cluster1" &
-cluster_clean_up "kind delete cluster --name cluster2" &
+if command_exists "k3d"; then
+    cluster_clean_up "k3d cluster delete kubeflex" &
+    cluster_clean_up "k3d cluster delete cluster1" &
+    cluster_clean_up "k3d cluster delete cluster2" &
+    wait
+fi
+if command_exists "kind"; then
+    cluster_clean_up "kind delete cluster --name kubeflex" &
+    cluster_clean_up "kind delete cluster --name cluster1" &
+    cluster_clean_up "kind delete cluster --name cluster2" &
+    wait
+fi
+
 wait
 echo -e "\033[33m✔\033[0m Cluster space clean up has been completed"
 
 echo -e "\nStarting context clean up..."
 context_clean_up
 echo -e "\033[33m✔\033[0m Context space clean up completed"
 
-echo -e "\nCreating two clusters to serve as example WECs"
+echo -e "\nCreating two $k8s_platform clusters to serve as example WECs"
 clusters=(cluster1 cluster2)
 cluster_log_dir=$(mktemp -d)
 trap "rm -rf $cluster_log_dir" EXIT
 for cluster in "${clusters[@]}"; do
-    kind create cluster --name "${cluster}" >"${cluster_log_dir}/${cluster}.log" 2>&1 && touch "${cluster_log_dir}/${cluster}.success" &
+    if [ "$k8s_platform" == "kind" ]; then
+        kind create cluster --name "${cluster}" >"${cluster_log_dir}/${cluster}.log" 2>&1 && touch "${cluster_log_dir}/${cluster}.success" &
+    else
+        k3d cluster create --network k3d-kubeflex "${cluster}" >"${cluster_log_dir}/${cluster}.log" 2>&1 && touch "${cluster_log_dir}/${cluster}.success" &
+        wait
+    fi
 done
 
 echo -e "Creating KubeFlex cluster with SSL Passthrough"
-curl -s https://raw.githubusercontent.com/kubestellar/kubestellar/v${kubestellar_version}/scripts/create-kind-cluster-with-SSL-passthrough.sh | bash -s -- --name kubeflex --nosetcontext
+if [ "$k8s_platform" == "kind" ]; then
+    curl -s https://raw.githubusercontent.com/kubestellar/kubestellar/v${kubestellar_version}/scripts/create-kind-cluster-with-SSL-passthrough.sh | bash -s -- --name kubeflex --nosetcontext
+else
+    k3d cluster create -p "9443:443@loadbalancer" --k3s-arg "--disable=traefik@server:*" kubeflex
+    sleep 15
+    helm install ingress-nginx ingress-nginx --set "controller.extraArgs.enable-ssl-passthrough=" --repo https://kubernetes.github.io/ingress-nginx --version 4.6.1 --namespace ingress-nginx --create-namespace
```

Comment:
I think there was an unintentional slip back to the older version of the ingress chart here.

Reply:
yep - was trying to see if it helped with the CSR stuff - ill put it back now